### PR TITLE
New filter panel layout

### LINKED ISSUE #1379: Implement new styles to filter panel and results area
So that users have a more consistent experience across the site, restyle the filter panel, tag population area, and content area on the data tables and calendar.

These changes will make these elements better match the panel on the [regulation viewer](https://beta.fec.gov/regulations/1).

Using the designs from [Visual design of co-located tag and filter system #1329](https://github.com/18F/openFEC-web-app/issues/1329#issuecomment-234105298)

**Criteria for completion:**
- [x] Add 1px `$primary` border right to filter panel for its full height
- [x] Move tag area between table title and table results, change background to white
- [x] Move "Export" button inline with table title, aligned right
- [x] Add de-activated "Share" button
- [x] On mobile, drop "Export" button
- [x] Add filter icon in `$primary` circle
 - [x] @jenniferthibault to export icon
- [x] Replace filter panel close icon with `.i-arrow-left-border`
- [x] Change filter panel title copy to "Edit filters"
- [x] Add 1px `$primary` border bottom to panel title area, 
  - This border should align with the border under the data table title on the right
- [x] Extend 2px `$primary` top border on table to touch the filter panel on left and edge of screen on right

On data tables implementation
- [x] Remove "View x results per page" functionality & interaction
- [x] Move table pagination buttons to bottom of table

On calendar implementation
- [x] Move month selection arrows to left of Month title
- [x] Move current month selection button to right of Month title, activate only when user is viewing a different month than current

All
- [x] Pair with @jenniferthibault for final style review on data tables and calendar implementation

### PR DESCRIPTION
**Pull request by noahmanger**
Implements the new designs for the filter panel area.

Requires https://github.com/18F/fec-style/pull/446
Resolves https://github.com/18F/openFEC-web-app/issues/1379

### CODE REVIEW

**Comment by noahmanger** [comment]
Ready for review.

File: package.json [hunk 1]
```diff
@@ -66,7 +66,7 @@
     "d3": "3.5.5",
     "datatables.net": "1.10.10",
     "datatables.net-responsive": "2.0.1",
-    "fec-style": "2.8.1",
+    "fec-style": "git://github.com/18f/fec-style#new-filter-panel",
```
Comment (adborden) [inline package.json]: Do you want to leave this in?

Comment (adborden) [inline package.json]: Actually, you can bump this later today once the other PR is merged :)

File: static/js/modules/tables.js [hunk 2]
```diff
@@ -591,6 +586,11 @@ DataTable.prototype.fetchSuccess = function(resp) {
 
   var changeCount = this.newCount - this.currentCount;
 
+  var countHTML = this.newCount > 0 ?
+    'about <span class="tags__count">' + this.newCount.toLocaleString('en-US') + '</span>' :
```
Comment (adborden) [inline static/js/modules/tables.js]: nice!

File: package.json (hunk 1 above)
Comment (noahmanger) [inline package.json]: Done.

**Pull request merged**